### PR TITLE
Tidy Up Constructors and Destructors

### PR DESCRIPTION
**Pull request by krishty**
This commit does not add or remove c’tors or d’tors, so it is *not* ABI-breaking.

If a c’tor/d’tor does nothing else than the default behavior, this commit replaces it with `= default`.

If an initializer list entry does nothing else than the default behavior, this commit removes it. First and foremost, remove default c’tor calls of base classes (always called by the compiler if no other base c’tor is explicitly called) and c’tor calls of members with complex types (e.g. `std::vector`).

In a few instances, user-defined copy c’tors / move c’tors / assignment operators / move assignment operators were replaced with `= default`, too. I only did this if I had a clear understanding of what’s going on.

### CODE REVIEW

File: code/AssetLib/Blender/BlenderScene.h [hunk 1]
```diff
@@ -729,7 +728,7 @@ struct Object : ElemBase {
     ListBase modifiers;
 
     Object() :
-            ElemBase(), type(Type_EMPTY), parent(nullptr), track(), proxy(), proxy_from(), data() {
+            type(Type_EMPTY), parent(nullptr) {
```
Comment (krishty) [inline code/AssetLib/Blender/BlenderScene.h:731]: base class default c’tors are always called; `std::shared_ptr` is always default-initialized

File: code/AssetLib/Blender/BlenderScene.h [hunk 2]
```diff
@@ -741,8 +740,7 @@ struct Base : ElemBase {
     std::shared_ptr<Object> object WARN;
 
     Base() :
-            ElemBase(), prev(nullptr), next(), object() {
-        // empty
```
Comment (krishty) [inline code/AssetLib/Blender/BlenderScene.h:745]: base class default c’tors are always called; `std::shared_ptr` is always default-initialized

File: code/AssetLib/Blender/BlenderScene.h [hunk 3]
```diff
@@ -758,10 +756,7 @@ struct Scene : ElemBase {
 
     ListBase base;
 
-    Scene() :
-            ElemBase(), camera(), world(), basact(), master_collection() {
-        // empty
-    }
```
Comment (krishty) [inline code/AssetLib/Blender/BlenderScene.h:764]: base class default c’tors are always called; `std::shared_ptr` is always default-initialized

File: code/AssetLib/Blender/BlenderScene.h [hunk 4]
```diff
@@ -791,10 +786,7 @@ struct Image : ElemBase {
 
     short gen_x, gen_y, gen_type;
 
-    Image() :
-            ElemBase() {
-        // empty
-    }
+    Image() = default;
```
Comment (krishty) [inline code/AssetLib/Blender/BlenderScene.h:789]: base class default c’tors are always called

File: code/AssetLib/Blender/BlenderScene.h [hunk 5]
```diff
@@ -884,7 +876,7 @@ struct Tex : ElemBase {
     //char use_nodes;
 
     Tex() :
-            ElemBase(), imaflag(ImageFlags_INTERPOL), type(Type_CLOUDS), ima() {
+            imaflag(ImageFlags_INTERPOL), type(Type_CLOUDS) {
```
Comment (krishty) [inline code/AssetLib/Blender/BlenderScene.h:879]: base class default c’tors are always called; `std::shared_ptr` is always default-initialized

File: code/AssetLib/Blender/BlenderScene.h [hunk 6]
```diff
@@ -976,10 +968,7 @@ struct MTex : ElemBase {
     //float shadowfac;
     //float zenupfac, zendownfac, blendfac;
 
-    MTex() :
-            ElemBase() {
-        // empty
-    }
+    MTex() = default;
```
Comment (krishty) [inline code/AssetLib/Blender/BlenderScene.h:971]: base class default c’tors are always called

File: code/AssetLib/C4D/C4DImporter.cpp [hunk 7]
```diff
@@ -106,15 +106,10 @@ static const aiImporterDesc desc = {
 
 
 // ------------------------------------------------------------------------------------------------
-C4DImporter::C4DImporter()
-: BaseImporter() {
-    // empty
-}
+C4DImporter::C4DImporter() = default;
```
Comment (krishty) [inline code/AssetLib/C4D/C4DImporter.cpp:109]: base class default c’tors are always called

File: code/AssetLib/Irr/IRRMeshLoader.cpp [hunk 8]
```diff
@@ -71,11 +71,7 @@ static const aiImporterDesc desc = {
 
 // ------------------------------------------------------------------------------------------------
 // Constructor to be privately used by Importer
-IRRMeshImporter::IRRMeshImporter() :
-        BaseImporter(),
-        IrrlichtBase() {
-    // empty
-}
+IRRMeshImporter::IRRMeshImporter() = default;
```
Comment (krishty) [inline code/AssetLib/Irr/IRRMeshLoader.cpp:74]: base class default c’tors are always called

File: code/AssetLib/glTF/glTFImporter.cpp [hunk 9]
```diff
@@ -80,7 +80,7 @@ static const aiImporterDesc desc = {
 };
 
 glTFImporter::glTFImporter() :
-        BaseImporter(), meshOffsets(), embeddedTexIdxs(), mScene(nullptr) {
+        mScene(nullptr) {
```
Comment (krishty) [inline code/AssetLib/glTF/glTFImporter.cpp:83]: base class default c’tors are always called; `std::vector` is always default-initialized

File: code/AssetLib/glTF2/glTF2Importer.cpp [hunk 10]
```diff
@@ -96,9 +96,6 @@ static const aiImporterDesc desc = {
 };
 
 glTF2Importer::glTF2Importer() :
-        BaseImporter(),
-        meshOffsets(),
-        mEmbeddedTexIdxs(),
```
Comment (krishty) [inline code/AssetLib/glTF2/glTF2Importer.cpp:101]: base class default c’tors are always called; `std::shared_ptr` is always default-initialized

File: code/AssetLib/Collada/ColladaLoader.cpp [hunk 11]
```diff
@@ -92,15 +92,6 @@ inline void AddNodeMetaData(aiNode *node, const std::string &key, const T &value
...
 ColladaLoader::ColladaLoader() :
-        mFileName(),
-        mMeshIndexByID(),
-        mMaterialIndexByName(),
-        mMeshes(),
-        newMats(),
-        mCameras(),
-        mLights(),
-        mTextures(),
-        mAnims(),
```
Comment (krishty) [inline code/AssetLib/Collada/ColladaLoader.cpp:103]: `std::map`, `std::string`, and `std::vector` are always default-initialized

File: code/AssetLib/DXF/DXFHelper.h [hunk 12]
```diff
@@ -65,7 +65,6 @@ class LineReader {
     LineReader(StreamReaderLE& reader)
     : splitter(reader,false,true)
     , groupcode( 0 )
-    , value()
```
Comment (krishty) [inline code/AssetLib/DXF/DXFHelper.h:68]: `std::string` is always default-initialized

File: code/AssetLib/DXF/DXFHelper.h [hunk 13]
```diff
@@ -186,8 +185,7 @@ struct InsertBlock {
     InsertBlock()
     : pos()
     , scale(1.f,1.f,1.f)
-    , angle()
-    , name() {
```
Comment (krishty) [inline code/AssetLib/DXF/DXFHelper.h:190]: `std::string` is always default-initialized

File: code/AssetLib/FBX/FBXExportNode.h [hunk 14]
```diff
@@ -77,8 +77,6 @@ class FBX::Node {
     /// The class constructor with the name.
     Node(const std::string& n)
     : name(n)
-    , properties()
-    , children()
```
Comment (krishty) [inline code/AssetLib/FBX/FBXExportNode.h:81]: `std::vector` is always default-initialized

File: code/AssetLib/FBX/FBXExportNode.h [hunk 15]
```diff
@@ -87,8 +85,6 @@ class FBX::Node {
     template <typename... More>
     Node(const std::string& n, More&&... more)
     : name(n)
-    , properties()
-    , children()
```
Comment (krishty) [inline code/AssetLib/FBX/FBXExportNode.h:91]: see above

File: code/AssetLib/FBX/FBXImporter.cpp [hunk 16]
```diff
@@ -90,10 +90,7 @@ static const aiImporterDesc desc = {
 
 // ------------------------------------------------------------------------------------------------
 // Constructor to be privately used by #Importer
-FBXImporter::FBXImporter() :
-        mSettings() {
```
Comment (krishty) [inline code/AssetLib/FBX/FBXImporter.cpp:94]: `FBX::ImportSettings` has a user-defined default c’tor and is therefore always default-initialized

File: code/AssetLib/OpenGEX/OpenGEXImporter.cpp [hunk 17]
```diff
@@ -261,7 +261,6 @@ OpenGEXImporter::RefInfo::RefInfo(aiNode *node, Type type, std::vector<std::stri
 OpenGEXImporter::OpenGEXImporter() :
         m_root(nullptr),
         m_nodeChildMap(),
-        m_meshCache(),
```
Comment (krishty) [inline code/AssetLib/OpenGEX/OpenGEXImporter.cpp:264]: `std::vector` is always default-initialized

File: code/AssetLib/Ply/PlyParser.h [hunk 18]
```diff
@@ -359,10 +357,7 @@ class PropertyInstance
 class ElementInstance {
 public:
     //! Default constructor
-    ElementInstance()  AI_NO_EXCEPT
-    : alProperties() {
```
Comment (krishty) [inline code/AssetLib/Ply/PlyParser.h:363]: `std::vector` is always default-initialized

File: code/AssetLib/Ply/PlyParser.h [hunk 19]
```diff
@@ -386,10 +381,7 @@ class ElementInstanceList
 public:
 
     //! Default constructor
-    ElementInstanceList() AI_NO_EXCEPT
-    : alInstances() {
```
Comment (krishty) [inline code/AssetLib/Ply/PlyParser.h:390]: `std::vector` is always default-initialized

File: code/AssetLib/Ply/PlyParser.h [hunk 20]
```diff
@@ -413,11 +405,7 @@ class DOM
 public:
 
     //! Default constructor
-    DOM() AI_NO_EXCEPT
-    : alElements()
-    , alElementData() {
```
Comment (krishty) [inline code/AssetLib/Ply/PlyParser.h:418]: `std::vector` is always default-initialized

File: code/AssetLib/Q3BSP/Q3BSPFileData.h [hunk 21]
```diff
@@ -169,19 +169,7 @@ struct Q3BSPModel {
...
-    Q3BSPModel() :
-        m_Data(),
-        m_Lumps(),
-        m_Vertices(),
-        m_Faces(),
-        m_Indices(),
-        m_Textures(),
-        m_Lightmaps(),
-        m_EntityData(),
-        m_ModelName()
```
Comment (krishty) [inline code/AssetLib/Q3BSP/Q3BSPFileData.h:181]: `std::vector` and `std::string` are always default-initialized

File: code/AssetLib/Q3BSP/Q3BSPFileImporter.cpp [hunk 22]
```diff
@@ -139,7 +139,7 @@ static void normalizePathName(const std::string &rPath, std::string &normalizedP
 // ------------------------------------------------------------------------------------------------
 //  Constructor.
 Q3BSPFileImporter::Q3BSPFileImporter() :
-        m_pCurrentMesh(nullptr), m_pCurrentFace(nullptr), m_MaterialLookupMap(), mTextures() {
+        m_pCurrentMesh(nullptr), m_pCurrentFace(nullptr) {
```
Comment (krishty) [inline code/AssetLib/Q3BSP/Q3BSPFileImporter.cpp:142]: `std::map` and `std::vector` are always default-initialized

File: code/AssetLib/SMD/SMDLoader.cpp [hunk 23]
```diff
@@ -83,7 +83,6 @@ static const aiImporterDesc desc = {
 // Constructor to be privately used by Importer
 SMDImporter::SMDImporter() :
         configFrameID(), 
-        mBuffer(),
```
Comment (krishty) [inline code/AssetLib/SMD/SMDLoader.cpp:86]: `std::vector` is always default-initialized

File: code/AssetLib/SMD/SMDLoader.h [hunk 24]
```diff
@@ -87,7 +87,7 @@ struct Vertex {
 */
 struct Face {
     Face() AI_NO_EXCEPT :
-            iTexture(0x0), avVertices{} {
```
Comment (krishty) [inline code/AssetLib/SMD/SMDLoader.h:90]: `Assimp::SMD::Vertex` has a user-defined default c’tor, which is always called

File: code/AssetLib/X/XFileHelper.h [hunk 25]
```diff
@@ -67,7 +67,6 @@ struct TexEntry {
     bool mIsNormalMap; // true if the texname was specified in a NormalmapFilename tag
 
     TexEntry() AI_NO_EXCEPT :
-            mName(),
```
Comment (krishty) [inline code/AssetLib/X/XFileHelper.h:70]: `std::string` is always default-initialized

File: code/AssetLib/X/XFileHelper.h [hunk 26]
```diff
@@ -128,17 +127,8 @@ struct Mesh {
 
     explicit Mesh(const std::string &pName = std::string()) AI_NO_EXCEPT
             : mName(pName),
-              mPositions(),
-              mPosFaces(),
-              mNormals(),
-              mNormFaces(),
```
Comment (krishty) [inline code/AssetLib/X/XFileHelper.h:134]: `std::vector` is always default-initialized

File: code/AssetLib/X/XFileHelper.h [hunk 27]
```diff
@@ -128,17 +127,8 @@ struct Mesh {
...
               mNumTextures(0),
-              mTexCoords{},
-              mNumColorSets(0),
-              mColors{},
-              mFaceMaterials(),
-              mMaterials(),
-              mBones() {
+              mNumColorSets(0) {
```
Comment (krishty) [inline code/AssetLib/X/XFileHelper.h:131]: `std::vector` is always default-initialized

File: code/AssetLib/X/XFileHelper.h [hunk 28]
```diff
@@ -152,15 +142,12 @@ struct Node {
     std::vector<Mesh *> mMeshes;
 
     Node() AI_NO_EXCEPT
-            : mName(),
```
Comment (krishty) [inline code/AssetLib/X/XFileHelper.h:155]: `std::string` is always default-initialized

File: code/AssetLib/X/XFileHelper.h [hunk 29]
```diff
@@ -152,15 +142,12 @@ struct Node {
...
-              mTrafoMatrix(),
-              mParent(nullptr),
-              mChildren(),
-              mMeshes() {
+            : mTrafoMatrix(),
+              mParent(nullptr) {
```
Comment (krishty) [inline code/AssetLib/X/XFileHelper.h:146]: `std::vector` is always default-initialized

File: code/AssetLib/X/XFileHelper.h [hunk 30]
```diff
@@ -152,15 +142,12 @@ struct Node {
...
         // empty
     }
     explicit Node(Node *pParent) :
-            mName(), mTrafoMatrix(), mParent(pParent), mChildren(), mMeshes() {
+            mTrafoMatrix(), mParent(pParent) {
```
Comment (krishty) [inline code/AssetLib/X/XFileHelper.h:150]: see above

File: code/AssetLib/X/XFileHelper.h [hunk 31]
```diff
@@ -211,8 +198,6 @@ struct Scene {
 
     Scene() AI_NO_EXCEPT
             : mRootNode(nullptr),
-              mGlobalMeshes(),
-              mGlobalMaterials(),
```
Comment (krishty) [inline code/AssetLib/X/XFileHelper.h:215]: `std::vector` is always default-initialized

File: code/AssetLib/X/XFileImporter.cpp [hunk 32]
```diff
@@ -75,9 +75,7 @@ static const aiImporterDesc desc = {
 
 // ------------------------------------------------------------------------------------------------
 // Constructor to be privately used by Importer
-XFileImporter::XFileImporter() : mBuffer() {
```
Comment (krishty) [inline code/AssetLib/X/XFileImporter.cpp:78]: `std::vector` is always default-initialized

File: code/AssetLib/X3D/X3DExporter.hpp [hunk 33]
```diff
@@ -52,22 +52,14 @@ class X3DExporter {
     struct SAttribute {
         const std::string Name;
         const std::string Value;
-        SAttribute() :
-                Name(),
-                Value() {
-            // empty
-        }
+        SAttribute() = default;
```
Comment (krishty) [inline code/AssetLib/X3D/X3DExporter.hpp:55]: `std::string` is always default-initialized

File: code/AssetLib/X3D/X3DExporter.hpp [hunk 34]
```diff
@@ -52,22 +52,14 @@ class X3DExporter {
...
                 Value(value) {
             // empty
         }
 
-        SAttribute(SAttribute &&rhs) AI_NO_EXCEPT :
-                Name(rhs.Name),
-                Value(rhs.Value) {
-            // empty
-        }
+        SAttribute(SAttribute &&rhs) AI_NO_EXCEPT = default;
```
Comment (krishty) [inline code/AssetLib/X3D/X3DExporter.hpp:62]: see above

File: code/AssetLib/glTF2/glTF2Asset.h [hunk 35]
```diff
@@ -365,16 +365,7 @@ struct CustomExtension {
 
     ~CustomExtension() = default;
 
-    CustomExtension(const CustomExtension &other) :
-            name(other.name),
```
Comment (krishty) [inline code/AssetLib/glTF2/glTF2Asset.h:369]: `std::string` is always default-initialized

File: code/AssetLib/glTF2/glTF2Asset.h [hunk 36]
```diff
@@ -365,16 +365,7 @@ struct CustomExtension {
...
-            mStringValue(other.mStringValue),
-            mDoubleValue(other.mDoubleValue),
-            mUint64Value(other.mUint64Value),
-            mInt64Value(other.mInt64Value),
-            mBoolValue(other.mBoolValue),
-            mValues(other.mValues) {
```
Comment (krishty) [inline code/AssetLib/glTF2/glTF2Asset.h:375]: `Nullable` has a user-defined c’tor and is therefore always default-initialized

File: code/AssetLib/glTF2/glTF2Asset.h [hunk 37]
```diff
@@ -1086,8 +1077,7 @@ struct AssetMetadata {
 
     void Read(Document &doc);
 
-    AssetMetadata() :
-            version() {}
```
Comment (krishty) [inline code/AssetLib/glTF2/glTF2Asset.h:1090]: `std::string` is always default-initialized

File: include/assimp/IOSystem.hpp [hunk 38]
```diff
@@ -237,10 +237,7 @@ class ASSIMP_API IOSystem
 };
 
 // ----------------------------------------------------------------------------
-AI_FORCE_INLINE IOSystem::IOSystem() AI_NO_EXCEPT :
-        m_pathStack() {
-    // empty
-}
+AI_FORCE_INLINE IOSystem::IOSystem() AI_NO_EXCEPT = default;
```
Comment (krishty) [inline include/assimp/IOSystem.hpp:240]: `std::vector` is always default-initialized

File: include/assimp/LineSplitter.h [hunk 39]
```diff
@@ -145,7 +145,6 @@ class LineSplitter {
 
 AI_FORCE_INLINE LineSplitter::LineSplitter(StreamReaderLE& stream, bool skip_empty_lines, bool trim ) :
         mIdx(0),
-        mCur(),
```
Comment (krishty) [inline include/assimp/LineSplitter.h:148]: `std::string` is always default-initialized

**Comment by kimkulling** [comment]
Merged, thanks a lot for your contribution.

**Pull request merged**